src/setup-beam.js

```diff
@@ -620,13 +653,26 @@ function parseVersionFile(versionFilePath0) {
       }
     }
   })
-  if (!appVersions.size) {
-    core.info('There was apparently nothing to consume')
-  } else {
-    core.info('... done!')
-  }
-  core.endGroup()
+  return appVersions
+}
 
+function parseMiseVersionFile(fileData) {
+  const appVersions = new Map()
+  const toml = TOML.parse(fileData)
+  for (let app of ['erlang', 'elixir', 'gleam', 'rebar']) {
+    let version = toml.tools && toml.tools[app]
+    if (Array.isArray(version)) {
```

Comment:
I'd add a comment here: "In case of multiple versions, only the first one is picked up".